source/Calamari.Aws/Kubernetes/Discovery/AwsAuthenticationDetails.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections.Generic;
+using Amazon.Runtime;
+using Calamari.Common.Features.Discovery;
+
+namespace Calamari.Kubernetes.Aws
+{
+    public class AwsAuthenticationDetails : ITargetDiscoveryAuthenticationDetails
+    {
+        const string DefaultSessionName = "OctopusKubernetesClusterDiscovery";
+        public AWSCredentials ToCredentials()
+        {
+            var account = Credentials.Type == "account"
+                ? new BasicAWSCredentials(Credentials.Account.AccessKey, Credentials.Account.SecretKey)
+                : (AWSCredentials)new EnvironmentVariablesAWSCredentials();
```

Comment:
I'm a bit worried that this might not handle the E2C Instance case (after talking in the Cloud Credentials meeting). We don't have a test using an E2C instance worker to test and se we're only using EnvironmentVariables in tests where `Credentials.Type` is not account...

Reply:
I've updated this now but we don't have any tests which exercise the `InstanceProfileAwsCredentials`... Should I add that in?